src/Chainweb/CutDB.hs

```diff
@@ -519,11 +520,11 @@ cutHashesToBlockHeaderMap
         -- a 'Cut'.
 cutHashesToBlockHeaderMap logfun headerStore payloadStore hs =
     trace logfun "Chainweb.CutDB.cutHashesToBlockHeaderMap" (_cutId hs) 1 $ do
-        hdrs <- emptyCas
-        traverse_ (casInsert hdrs) $ _cutHashesHeaders hs
-
         plds <- emptyCas
-        traverse_ (casInsert plds) $ _cutHashesPayloads hs
+        casInsertBatch plds $ V.fromList $ HM.elems $ _cutHashesPayloads hs
+
+        hdrs <- emptyCas
+        casInsertBatch hdrs $ V.fromList $ HM.elems $ _cutHashesHeaders hs
```

Comment:
How many items do we expect to insert at once? Could this ever be an expensive `Vector` conversion? GHC might be nice and fuse the inner list away anyway.

Reply:
Agreed, using vector in the class API was probably a mistake. Lazy `[]` would have been more flexible. But changing the `IsCas` class was beyond the scope of this PR.

In any case in practice this shouldn't make a big difference. Batches shouldn't be long in any case. Not sure, though if GHC is smart enough to scope the allocation for the vector.

Reply:
Usually it is in the case of pulling keys lists from a `Map`.